modules/apps/transfer/keeper/relay_test.go

```diff
@@ -18,137 +17,132 @@ import (
+			coin = sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(100))
+			sender = suite.chainA.SenderAccount.GetAddress()
+			timeoutHeight = suite.chainB.GetTimeoutHeight()
 
-			if !tc.sendFromSource {
-				// send coin from chainB to chainA
-				coinFromBToA := sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(100))
-				transferMsg := types.NewMsgTransfer(path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID, coinFromBToA, suite.chainB.SenderAccount.GetAddress().String(), suite.chainA.SenderAccount.GetAddress().String(), clienttypes.NewHeight(1, 110), 0)
-				_, err = suite.chainB.SendMsgs(transferMsg)
-				suite.Require().NoError(err) // message committed
+			// create IBC token on chainA
+			transferMsg := types.NewMsgTransfer(path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID, coin, suite.chainB.SenderAccount.GetAddress().String(), suite.chainA.SenderAccount.GetAddress().String(), suite.chainA.GetTimeoutHeight(), 0)
+			result, err := suite.chainB.SendMsgs(transferMsg)
+			suite.Require().NoError(err) // message committed
 
-				// receive coin on chainA from chainB
-				fungibleTokenPacket := types.NewFungibleTokenPacketData(coinFromBToA.Denom, coinFromBToA.Amount.String(), suite.chainB.SenderAccount.GetAddress().String(), suite.chainA.SenderAccount.GetAddress().String())
-				packet := channeltypes.NewPacket(fungibleTokenPacket.GetBytes(), 1, path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID, path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, clienttypes.NewHeight(1, 110), 0)
+			packet, err := ibctesting.ParsePacketFromEvents(result.GetEvents())
+			suite.Require().NoError(err)
 
-				// get proof of packet commitment from chainB
-				err = path.EndpointA.UpdateClient()
-				suite.Require().NoError(err)
-				packetKey := host.PacketCommitmentKey(packet.GetSourcePort(), packet.GetSourceChannel(), packet.GetSequence())
-				proof, proofHeight := path.EndpointB.QueryProof(packetKey)
+			err = path.RelayPacket(packet)
+			suite.Require().NoError(err)
 
-				recvMsg := channeltypes.NewMsgRecvPacket(packet, proof, proofHeight, suite.chainA.SenderAccount.GetAddress().String())
-				_, err = suite.chainA.SendMsgs(recvMsg)
-				suite.Require().NoError(err) // message committed
-			}
+			tc.malleate()
 
-			err = suite.chainA.GetSimApp().TransferKeeper.SendTransfer(
-				suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, amount,
-				sender, suite.chainB.SenderAccount.GetAddress().String(), suite.chainB.GetTimeoutHeight(), 0,
+			msg := types.NewMsgTransfer(
+				path.EndpointA.ChannelConfig.PortID,
+				path.EndpointA.ChannelID,
+				coin, sender.String(), suite.chainB.SenderAccount.GetAddress().String(),
+				timeoutHeight, 0, // only use timeout height
 			)
 
+			res, err := suite.chainA.GetSimApp().TransferKeeper.Transfer(sdk.WrapSDKContext(suite.chainA.GetContext()), msg)
+
 			if tc.expPass {
 				suite.Require().NoError(err)
+				suite.Require().NotNil(res)
```

Comment:
Do you think we should also add here checks for changes that we expect to have happened in case of success (e.g. the expected amount is escrowed in the escrow account)? We check for this already in the e2e tests but maybe it's good to add here for redundancy?

Same comment maybe for the failure cases.

Reply:
I think we should. But maybe we should do in a followup issue so as not to increase the scope of this one too much?